### ISSUE TITLE
Rls stuck at "installing components..." vs code

### ISSUE THREAD

**Issue by seaofsole** [rust-lang/rls-vscode#663]
I am new to rust and installing it to vscode, OS is windows things I have done to:
installing rustup
ran "rustc --version" gave me a version number, so i assumed all is well 

now when to vscode to begin playing around with the language steps i took:
-cd into the a newly created folder which i would use for rust development
- ran cargo run (this created a new project with the name i specified and initialized a git repo which was nice)

now the meat of the problem, when i installed the rls extension on vscode, it gave me an error "c:/program cannot be executed" something like that after i saw this little spinning wheel that says "RLS installing components" what do i do from here

I apologize if this is messy im new to rust normally python is a lot easier to set up and go with its linting, but i cant get that to work on rust using vs code, makes me think I'm doing something wrong.

also im new to github pardon me, thanks for your response

**Comment by barryblando** [comment]
I got it working when using VSCode WSL Remote. If you're using Windows 10, VSCode suddenly breaks the task, I still don't know why that happens it doesn't show any output why it breaks but VS Code RLS depends on native windows install when you're seeing this command "**rustup component add rust-analysis --toolchain stable-x86_64-pc-windows-msvc**" and when you're using WSL Remote this is what you will see:

![Annotation 2019-09-22 160113](https://user-images.githubusercontent.com/5582213/65384223-381d1e80-dd52-11e9-82da-54fb4eed93a3.png)

**Comment by SagnikPradhan** [comment]
![Capture](https://user-images.githubusercontent.com/26499576/65776978-bddd0780-e160-11e9-817c-1cf8443cf23b.PNG)
I am also facing a similar issue. It seems to be stuck at installing components.

**Comment by Xanewok** [comment]
I have to admit that the component installing feature was implemented before support for remote workspace (e.g. WSL, ssh) was implemented in VSCode, so there's a chance these don't interact properly. I'll look into it.

**Comment by tomaster777** [comment]
I was also facing the same issue on OSX. Manually setting "rust-client.rlsPath" to the output of ```which rls``` fixed it for me.

**Comment by JamesJGoodwin** [comment]
@Xanewok any updates on this one? Got stuck with `Installing components` on Windows 10 and VSC 1.44.0 and I don't have WSL extension installed.

**Comment by Vollkornaffe** [comment]
Stuck on Installing components, Ubuntu 18.04
got code via apt install

**Comment by JamesJGoodwin** [comment]
Yes, thank you @Vollkornaffe for mentioning here that thread. Manually installing RLS via: `rustup component add rls` fixes the issue. Yet it's not fixing this issue for those who are trying to install VSCode extension from the Marketplace.

**Comment by timichal** [comment]
I have the same issue on Windows (no WSL) and manually installing RLS doesn't help either - it's in PATH after the installation, but a prompt _RLS not installed. Install?_ still appears and the installation then gets stuck at _Installing components..._ again.

**Comment by qwefgh90** [comment]
I had the same issue on windows with mingw. It's been solved after manually installing `rustup component add rls rust-analysis rust-src` in https://github.com/rust-lang/rls. And it's done without `rlsPath`.

**Comment by hvram1** [comment]
This occurs if you use the nightly channel . (i.e.) rustup override set nightly . If i change it back to stable VScode is unable to detect it and still tries to download the nightly toolchain.
